### PR TITLE
Introduce new preRequest event and emit request on request event

### LINKED ISSUE #209: Shouldn't request event emit request object?
I'd expect for the `request` event to include the created request object, not simply the `uri` and `options` (which also allows mutations of these `options` resulting in potential side effects).

Similar to the functionality of the `response` event.

This would allow for consistent instrumentation of both `request` and `response` objects (attaching event listenings for logging, etc).

Instead, the only way to get at it today is through the promise's `req` property which doesn't feel ideal (but probably workable).

### PR DESCRIPTION
**Pull request by tlivings**
Closes #209

### CODE REVIEW

**Comment by geek** [comment]
@tlivings thoughts on keeping `request` how it was and we can say it's deprecated in the readme. Then where you have `request` now (after the request) we can rename to `onPostRequest`. Also, should we name it `onPreRequest`?

**Comment by tlivings** [comment]
`on(‘onPreRequest’ ...)` 🤔 seems redundant but sure.

My only problem with `onPostRequest` is that it sounds like after a request is made.

**Comment by geek** [comment]
Ya, I don't know what I was thinking. The name you have is good.

**Comment by geek** [comment]
@tlivings I'm going to merge as an enhancement... I'll move the existing `request` alongside `preRequest` so that it's not a breaking change. Then mark it as deprecated in the readme and recommend people to use the `preRequest` event.

File: lib/index.js [hunk 1]
```diff
@@ -183,13 +183,18 @@ internals.Client.prototype._request = function (method, url, options, relay, _tr
         uri.ciphers = options.ciphers;
     }
 
+    const start = Date.now();
```
Comment (geek) [inline lib/index.js]: I'd like to keep this where it was before so that the timing is only around the request and not the event handler.

**Pull request merged**